### PR TITLE
ECOM without payment gateway and enable course certificates

### PR DESCRIPTION
**Pull request by iam-mhaseeb**
**Description:** Restricted paid mode courses options for honor mode only customers

**Jira:** https://edlyio.atlassian.net/browse/EDLY-2092

**Related PR:** https://github.com/edly-io/edly-edx-themes/pull/218

**Screenshots:**
Screenshot 2020-10-28 at 1 46 19 PM
Screenshot 2020-10-28 at 1 46 24 PM
Screenshot 2020-10-28 at 1 46 30 PM

### CODE REVIEW

File: ecommerce/static/js/views/course_form_view.js [hunk 1]
```diff
@@ -158,22 +158,28 @@ define([
...
-                    activeCourseTypes = ['credit'];
-                    break;
-                default:
-                    activeCourseTypes = ['audit', 'verified', 'professional', 'credit'];
-                    break;
+                var disablePaidCourseModes = $('#disable_paid_course_modes').attr('value');
+                if (disablePaidCourseModes === true || disablePaidCourseModes === 'True') {
+                    activeCourseTypes = ['audit'];
+                }
+                else{
```
Comment (rehan99000) [inline ecommerce/static/js/views/course_form_view.js:165]: add a space between `else` and `{`

**Review by rehan99000 — APPROVED** [review]
just a minor format fix

File: ecommerce/core/templatetags/core_extras.py [hunk 2]
```diff
@@ -17,6 +17,23 @@ def settings_value(name):
     return getattr(settings, name)
 
 
+@register.simple_tag(takes_context=True)
+def site_configuration_value(context, name):
```
Comment (zubair-arbi) [inline ecommerce/core/templatetags/core_extras.py:21]: Move this to `ecommerce/extensions/edly_ecommerce_app`

File: ecommerce/static/js/views/course_form_view.js (hunk 1 above)
Comment (iam-mhaseeb) [reply ecommerce/static/js/views/course_form_view.js]: Done

File: ecommerce/templates/courses/course_app.html [hunk 3]
```diff
@@ -15,6 +16,7 @@
 
 {% block content %}
     <div id="app" class="container" data-credit-providers="{{ credit_providers }}"></div>
+    <input type="hidden" id="disable_paid_course_modes" name="disable_paid_course_modes" value="{% site_configuration_value "DISABLE_PAID_COURSE_MODES"  %}">
```
Comment (zubair-arbi) [inline ecommerce/templates/courses/course_app.html:19]: I would be great if we can override this template for our themes.

**Review by zubair-arbi — CHANGES_REQUESTED** [review]
I have added my review.

File: ecommerce/settings/_oscar.py [hunk 4]
```diff
@@ -18,6 +18,7 @@
     'ecommerce.extensions.api',
     'ecommerce.extensions.fulfillment',
     'ecommerce.extensions.refund',
+    'ecommerce.extensions.edly_ecommerce_app',
```
Comment (asadmanzoor93) [inline ecommerce/settings/_oscar.py:21]: why have we changed the position of `edly_ecommerce_app`?

File: ecommerce/static/js/views/course_form_view.js [hunk 5]
```diff
@@ -158,22 +158,28 @@ define([
...
-                    activeCourseTypes = ['professional'];
-                    break;
-                case 'credit':
-                    activeCourseTypes = ['credit'];
-                    break;
-                default:
-                    activeCourseTypes = ['audit', 'verified', 'professional', 'credit'];
-                    break;
+                var disablePaidCourseModes = $('#disable_paid_course_modes').attr('value');
+                if (disablePaidCourseModes === true || disablePaidCourseModes === 'True') {
```
Comment (asadmanzoor93) [inline ecommerce/static/js/views/course_form_view.js:162]: what if disablePaidCourseModes is equal to **'true'**. We should add a generic check-like comparing it with string after converting to lower and try to reduce this change to a single check.

**Review by asadmanzoor93 — CHANGES_REQUESTED** [review]
I have a question and one suggestion. Kindly review

File: ecommerce/static/js/views/course_form_view.js (hunk 5 above)
Comment (iam-mhaseeb) [reply ecommerce/static/js/views/course_form_view.js:162]: Yep, it's already on path. we would only get disablePaidCourseModes === 'True'. Updating code.

File: ecommerce/settings/_oscar.py (hunk 4 above)
Comment (iam-mhaseeb) [reply ecommerce/settings/_oscar.py:21]: With core apps, it wasn't allowing us to register template tags.

File: ecommerce/extensions/edly_ecommerce_app/tests/test_templatetags.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+
+from django.template import Context, Template
+
+from ecommerce.tests.testcases import TestCase
+
+
+class EdlyExtrasTests(TestCase):
+
+    def setUp(self):
```
Comment (rehan99000) [inline ecommerce/extensions/edly_ecommerce_app/tests/test_templatetags.py:10]: I think we don't need to override `setUp` if we are not declaring anything in it.

**Review by rehan99000 — APPROVED** [review]
just one suggestion

File: ecommerce/extensions/edly_ecommerce_app/tests/test_templatetags.py (hunk 6 above)
Comment (iam-mhaseeb) [reply ecommerce/extensions/edly_ecommerce_app/tests/test_templatetags.py:10]: I would keep it for future use to minimize line changes.

**Review by asadmanzoor93 — APPROVED** [review]

File: ecommerce/core/templatetags/core_extras.py (hunk 2 above)
Comment (iam-mhaseeb) [reply ecommerce/core/templatetags/core_extras.py]: Done

File: ecommerce/templates/courses/course_app.html (hunk 3 above)
Comment (iam-mhaseeb) [reply ecommerce/templates/courses/course_app.html]: Done

**Review by zubair-arbi — APPROVED** [review]
Changes looks good now. 👍 🚢 
Thanks for your support @rehan99000

**Pull request merged**